src/Domain/EmbeddingMaps/ProductMaps.hpp

```diff
@@ -0,0 +1,118 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines the class templates ProductOf2Maps and ProductOf3Maps.
+
+#pragma once
+
+#include <memory>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "EmbeddingMap.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+
+namespace EmbeddingMaps {
+
+/// Product of two codimension=0 EmbeddingMaps.
```

Comment:
Apply this to all classes being added (please check Identity and the abstract base class too :) )